### PR TITLE
Compute measure

### PR DESCRIPTION
**Pull request by wmatray**
Adds max flops to config file, default option is null which functions as normal

Uses ptflops package to calculate each model's operations count

General formula:

number of batches = max flops / (flops per model * batch size)

Requires #17

### CODE REVIEW

File: src/nn/utils.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    input_dtype = example_input.dtype
+    def input_constructor(input_shape):
+        return torch.zeros((1, *input_shape), dtype=input_dtype)
+    model_for_flops = copy.deepcopy(model)
+    flops, params = get_model_complexity_info(
+        model_for_flops, input_shape,
+        as_strings=False, print_per_layer_stat=False, verbose=False,
+        input_constructor=input_constructor
+    )
+    flops_per_batch = flops * batch_size * 2  # Forward + backward
```
Comment (MortimerE) [inline src/nn/utils.py:31]: Regarding my other comments, maybe I misunderstood - are flops in this context floating point operations period or floating point operations per second? In this case the estimator returns operations for one whole forward and backpass based on the operations required to compute a single batch, not an estimate of operations per second

File: src/gpt_evolution/run.py [hunk 2]
```diff
@@ -114,20 +115,40 @@ def configure_logger(experiment_path, logging_config):
...
 
+    # Check if dataset configuration is provided
+    if not all(key in tokenizer_config for key in ["dataset", "dataset_name"]):
+        logging.error("Dataset configuration must include 'dataset' and 'dataset_name'")
+        raise ValueError("Dataset configuration must include 'dataset' and 'dataset_name'")
+
     load_dataset_constant_kwargs = {"path": tokenizer_config["dataset"], "name": tokenizer_config["dataset_name"], "streaming": True}
+    
     if "data_files_prefixes" in tokenizer_config:
+        if "validation" not in tokenizer_config["data_files_prefixes"] or not tokenizer_config["data_files_prefixes"]["validation"]:
```
Comment (MortimerE) [inline src/gpt_evolution/run.py:126]: Some of this diff is also in #17 right?

**Review by MortimerE — APPROVED** [review]
This looks like a good functional implementation of the approach we discussed during meetings where we provide a compute budget and let the model run up until the point that compute is met. I think there are some optimizations we could make such as a lookback window (right now the cutoff could result in worse loss than a previous step), or a generally more robust approach to penalizing complexity, but this should accomplish our goals for the time being, and we can approach those ideas in downstream iterations of the codebase.
